src/components/container/MyDataHelpsAssistant/MyDataHelpsAssistant.tsx

```diff
@@ -0,0 +1,161 @@
+
+    const assistantRef = useRef<MyDataHelpsAssistant>();
+    const logRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (assistantRef.current === undefined) {
+            assistantRef.current = new MyDataHelpsAssistant(props.additionalInstructions, props.tools, props.appendTools);
+        }
+    }, []);
+
+    useEffect(() => {
+        if (logRef.current) {
+            logRef.current.scrollTop = logRef.current.scrollHeight;
+        }
+    }, [messages]);
+
+    const addUserMessage = async function () {
+
+        let newMessage = currentUserMessage;
+        setMessages(prevMessages => [...prevMessages, { type: 'user', content: newMessage }]);
+
+        setCurrentUserMessage('');
+
+        await assistantRef.current?.ask(newMessage, function (streamEvent: StreamEvent) {
+
+            const [kind, type] = streamEvent.event.split("_").slice(1);
+
+            if (type === "stream" && kind !== "chain") {
+                const chunk = streamEvent.data?.chunk;
+                let msg = chunk.message as AIMessageChunk;
+
+                if (msg.content && typeof msg.content === "string") {
+                    addMessageChunk(streamEvent.run_id, msg.content);
+                }
+                else if (props.debug && msg.tool_call_chunks && msg.tool_call_chunks.length > 0) {
+                    if (msg.tool_call_chunks[0].args) {
+                        addMessageChunk(streamEvent.run_id, msg.tool_call_chunks[0].args);
+                    }
+                    else if (msg.tool_call_chunks[0].name) {
+                        addMessageChunk(streamEvent.run_id, msg.tool_call_chunks[0].name + " ");
+                    }
+                }
+            }
+
+            if (kind === "tool") {
+                if (type === "start") {
+                    setLoading(`Querying your data...`);
+                }
+                else if (type === "end") {
+                    setLoading("");
+                }
+            }
+        });
+    }
+
+    const addMessageChunk = function (runId: string, message: string) {
+        setMessages(prevMessages => {
+            let existingMessage = prevMessages.find((msg) => msg.runId === runId);
+            if (existingMessage) {
+                const updatedMessage = { ...existingMessage, content: existingMessage.content + message };
+                return prevMessages.map(msg => msg.runId === runId ? updatedMessage : msg);
+            }
+            else {
+                return [...prevMessages, { type: 'ai', content: message, runId: runId }];
+            }
+        });
+    }
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setCurrentUserMessage(event.target.value);
+    }
+
+    const handleExpandClick = async () => {
+        setCollapsed(false);
+    }
+
+    return <>
+        {collapsed && <div className="mdh-assistant-collapsed" onClick={handleExpandClick}><FontAwesomeSvgIcon icon={faFlask} size="lg" /></div>}
```

Comment:
We're mixing a few things into this component; it's nice that it can just be a "drop in" but it's not great to be mixing the presentation logic into the thing itself; it envisions one specific way that this would be invoked via the icon on the bottom right.  Someone might want a different icon on the bottom right or that might not play well with their UI etc.  It's generally inconsistent with how views are presented in MDH; for something like this on mobile one would want to present a full screen chat window anyways presented via MyDataHelps.openApplication.  Generally we want to keep things like buttons here pretty generic

I would remove the button for invoking this entirely; this widget should JUST be the chat window itself.  

Later when we go to incorporate this in view builder, the chat window itself would be the component that would get dropped in.  This would allow for a lot of flexibility:
- Users could invoke the chat window from any button in their view; either modally or as a popup
- Or the chat window could just be a tab in their app

If we want to separately create a "Floating button" or "sticky button" component that could also be added to view builder, we can do that as well, but my 2c is that it is not necessary for this PR.